experimenter/experimenter/experiments/models.py

```diff
@@ -1347,6 +1353,124 @@ def get_branch_data(self, analysis_basis, selected_segment):
+                    .get(reference_branch, {})
+                    .get("all", [])
+                )
+
+                absolute_data_list.sort(key=self.formattedAnalysisPointComparator)
+                relative_data_list.sort(key=self.formattedAnalysisPointComparator)
+
+                significance_map = (
+                    metric_src.get("significance", {})
+                    .get(reference_branch, {})
+                    .get(window, {})
+                )
+
+                abs_entries = []
+                for i, data_point in enumerate(absolute_data_list):
+                    lower = round(data_point.get("lower"), 2)
+                    upper = round(data_point.get("upper"), 2)
+
+                    if metric.get("display_type") == "percentage":
+                        lower = f"{lower * 100}%"
+                        upper = f"{upper * 100}%"
+
+                    significance = significance_map.get(str(i + 1), "neutral")
+                    abs_entries.append(
+                        {"lower": lower, "upper": upper, "significance": significance}
+                    )
+
+                rel_entries = []
+                for i, data_point in enumerate(relative_data_list):
+                    lower = data_point.get("lower")
+                    upper = data_point.get("upper")
+                    interval = abs(round((lower + upper) / 2 * 100))
+                    significance = significance_map.get(str(i + 1), "neutral")
+                    rel_entries.append(
+                        {
+                            "lower": lower,
+                            "upper": upper,
+                            "significance": significance,
+                            "interval": interval,
+                        }
+                    )
+
+                branch_metrics[slug] = {"absolute": abs_entries, "relative": rel_entries}
+
+            metric_data[branch_slug] = branch_metrics
+
+        return metric_area_data
+
+    def get_metric_data(
+        self, analysis_basis, segment, reference_branch, window="overall"
+    ):
+        metric_areas = self.metric_areas
+        metric_data = {}
+
+        for area in metric_areas:
+            match area:
+                case "KPI Metrics":
+                    kpi_metric_list = self.get_metric_area_data(
+                        NimbusConstants.KPI_METRICS,
+                        analysis_basis,
+                        segment,
+                        reference_branch,
+                        window,
+                    )
+                    metric_data[area] = kpi_metric_list
+                case _:
+                    continue
+
+        return metric_data
+
+    @property
+    def sorted_branches(self):
+        branches = list(self.branches.all().order_by("slug"))
+        if self.reference_branch:
+            branches.remove(self.reference_branch)
+            branches.insert(0, self.reference_branch)
+        return branches
```

Comment:
Ah yeah there it is. 

Branches are default sorted by slug:

https://github.com/mozilla/experimenter/blob/main/experimenter/experimenter/experiments/models.py#L2142

And reference_branch will always be not-None by the time the experiment is created.

So you could just do something like

```py
return [self.reference_branch, *self.branches.exclude(id=self.reference_branch.id)]
```

instead of mutating a list.

Also it's mostly just a stylistic choice wehther to make something a property or not, but in the way it's used above, it's confusing to read becuase it's really a method call, which can be expensive/side-effectful, but it looks like a property access, which generally isn't.  So maybe this doesn't need to be a property?  Not a big deal, just something to think about.